ingest/src/main/scala/hydra.ingest/http/IngestionEndpoint.scala

```diff
@@ -72,8 +91,45 @@ class IngestionEndpoint[F[_]: Futurable](
           } ~ deleteRequest
         }
       }
+    } ~ pathPrefix("v2" / "topics") {
+      handleExceptions(exceptionHandler) {
+        pathSuffix("records") {
+            put {
+              pubV2
+            }
+        } ~ extractExecutionContext { implicit ec => getTopics }
+      }
     }
 
+  private val showSystemTopics = applicationConfig
+    .getBooleanOpt("transports.kafka.show-system-topics")
+    .getOrElse(false)
+
+  private val filterSystemTopics = (t: String) =>
+    (t.startsWith("_") && showSystemTopics) || !t.startsWith("_")
+
+
+  private def topics(implicit ec: ExecutionContext) : Future[Map[String, Seq[PartitionInfo]]] = {
+    implicit val timeout = Timeout(5 seconds)
+    implicit val cache = GuavaCache[Map[String, Seq[PartitionInfo]]]
+    cachingF("topics")(ttl = Some(1.minute)) {
+      import akka.pattern.ask
+      (consumerProxy ? ListTopics).mapTo[ListTopicsResponse].map { response =>
+        response.topics.filter(t => filterSystemTopics(t._1)).map {
+          case (k, v) => k -> v.toList
+        }
+      }
+    }
+  }
+
+  private def getTopics(implicit ec: ExecutionContext) = get {
+    handleExceptions(exceptionHandler) {
+      get {
+        complete(topics.map(_.keys))
+        }
+      }
+  }
```

Comment:
There appear to be two `get`s here. Guessing that's redundant?

Reply:
It is, will change.